### ISSUE TITLE
[UI Builder] Export Fails With Duplicate Name - Can't Change Export Name

### ISSUE THREAD

**Issue by JoelProminic** [prominic/Moonshine-IDE#69]
I ran a quick test with the Visual Editor today, and I found that I could not export the Flex Desktop project.

1. Open Moonshine
2. Click Visual Editor Project (Flex) on the home page
3. Name the project "TestVisualEditorProject"
4. Create the project
5. Add a simple Window and a label
6. Save the project
7. Go to Project > Export Visual Editor Project to Flex
8. BUG?:  The project name is not editable
9. Click Export
10. BUG:  Moonshine reports:  "This diretory already exists.  Please select a different path or project name."

![image](https://user-images.githubusercontent.com/24532731/32971067-327be18c-cbb1-11e7-82ea-c7001ee5bbe6.png)

**Comment by piotrzarzycki21** [comment]
I just fixed logic for this one and now user will be able to change name of exported project. Take a look once you get a chance!

**Comment by JoelProminic** [comment]
I was able to export the project, but the generated code did not match what I had created in the visual editor.

I created a simple application with a window and a single label:
![image](https://user-images.githubusercontent.com/24532731/33095150-3981aea2-cec8-11e7-9cdd-55661b949cca.png)

If I checked the code tab, I saw this:
```
?xml version="1.0" encoding="utf-8"?>
<s:WindowedApplication minWidth="955" minHeight="600" xmlns:fx="http://ns.adobe.com/mxml/2009" xmlns:s="library://ns.adobe.com/flex/spark" xmlns:mx="library://ns.adobe.com/flex/mx">
  
  <s:Panel x="159" y="94" width="200" height="200" title="Window">
    <s:Label x="7" y="12.5" width="189" height="15" text="HelloWorld"/>
  </s:Panel>
</s:WindowedApplication>
```


However, the exported project was a blank window:

```
<?xml version="1.0" encoding="utf-8"?>
<s:WindowedApplication xmlns:fx="http://ns.adobe.com/mxml/2009"
			   xmlns:s="library://ns.adobe.com/flex/spark" 
			   xmlns:mx="library://ns.adobe.com/flex/mx" minWidth="955" minHeight="600">
</s:WindowedApplication>
```

**Comment by piotrzarzycki21** [comment]
What was your steps ? - Exactly the same as in original description ?

**Comment by piotrzarzycki21** [comment]
And maybe a bit stupid question - Did you save the project before export ?

**Comment by JoelProminic** [comment]
Yes, the process was basically the same, and I saved the project.
1. Open Moonshine
2. Click Visual Editor Project (Flex) on the home page
3. Name the project "TestVisualEditorProject"
4. Create the project
5. Add a simple window and a label, as shown in my above screenshot.
6. Save the project with CMD-S
7. Go to Project > Export Visual Editor Project to Flex
8. Change the name
9. Click Export
10. Check the generated MXML file - it shows the window only.

By the way, I noticed that the visual editor window does not show an asterisk in the tab to indicate that there are unsaved changes.  I did confirm that the changes were saved by reopening the project, though.

**Comment by piotrzarzycki21** [comment]
@JoelProminic I got it and I know why it was working for me. Once you create new Visual Editor project, add some component and save - in the mxml file won't be written anything. If you go to the "Code" it look like everything is ok, but when you go physically on drive to that file it's empty. 

However once you add new component, go to the "Code" and Save - Export will properly work. I haven't noticed that probably cause whenever I saved project I have been doing it being on "Code" tab. 
I need to figure out why Moonshine is not saving file when we are on Visual tab.
